tests/test_basic.py

```diff
@@ -0,0 +1,50 @@
+import json
+import os
+
+
+def test_oda_api_code(dispatcher_live_fixture):
+    from oda_api.api import DispatcherAPI
+    disp = DispatcherAPI(
+        url=dispatcher_live_fixture
+    )
+
+    disp.get_product(
+        product_type="Dummy",
+        instrument="empty",
+        product="dummy"
+    )
+
+    job_id = disp.job_id
+    session_id = disp.session_id
+    scratch_path = f'scratch_sid_{session_id}_jid_{job_id}'
+    scratch_path_aliased = f'scratch_sid_{session_id}_jid_{job_id}_aliased'
+    assert os.path.exists(scratch_path) or os.path.exists(scratch_path_aliased)
+
+    if os.path.exists(scratch_path):
+        f_query_output = open(scratch_path + '/query_output.json')
+    else:
+        f_query_output = open(scratch_path_aliased + '/query_output.json')
+
+    jdata_output = json.load(f_query_output)
+
+    assert 'prod_dictionary' in jdata_output
+    assert 'api_code' in jdata_output['prod_dictionary']
+    output_api_code = jdata_output['prod_dictionary']['api_code']
+
+    expected_api_code = '''from oda_api.api import DispatcherAPI
+disp=DispatcherAPI(url='http://0.0.0.0:8001', instrument='mock', protocol='https')
```

Comment:
oh protocol is no more necessary. especially if it mismatches the url

Reply:
is it always provided within the url?

Reply:
it's not user, except it's validated. It should be left for compatibility, but it makes no sense, similarly to host.
URL by definintion contains protocol, host, port, path (as well as possibly credentials). The rest was some misunderstanding which eventually caused issues and was replaced.